client/incus_oidc.go

```diff
@@ -147,6 +148,40 @@ func (o *oidcClient) do(req *http.Request) (*http.Response, error) {
 	return resp, nil
 }
 
+// dial function executes a websocket request and handles OIDC authentication and refresh.
+func (o *oidcClient) dial(dialer websocket.Dialer, uri string, req *http.Request) (*websocket.Conn, *http.Response, error) {
+	conn, resp, err := dialer.Dial(uri, req.Header)
+	if err != nil && resp == nil {
```

Comment:
So what if err != nil but resp != nil ?  You will ignore the error in that case.  Is that the right thing?

Reply:
Yeah because that's specifically the case we want to retry.

Reply:
The special logic in this function is to handle cases where we get err != nil (Bad handshake) but have a resp filled in with the response code (403 - Unauthorized) and headers (OIDC issuer and client id).

When that happens, we refresh the OIDC token and then re-issue the dialer.Dial with the new token.